myokit/_model_api.py

```diff
@@ -4357,10 +4359,11 @@ def lhs(self):
         """
         return self._lhs
 
-    def promote(self, state_value=0):
+    def promote(self, state_value=0, check_const=True):
```

Comment:
I think I'd prefer to leave this to the validation stage? So have the checks in `validate` only.

I've tried a few times to make one of those APIs that doesn't let you construct a bad model, but it makes it very tiresome to use!

Reply:
so I've left in the constant check, as I think it is needed if a user uses promote and passes a non-constant expression for `state_value`

Reply:
but I've removed the `check_const` flag, as now the parser calls `promote` without `state_value` (i.e. now promote does not support proto expressions)